### PR TITLE
 Mock support for "Get Partitions Assigned to Adapter"

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.

### CODE REVIEW

**Review by anilak11 — CHANGES_REQUESTED** [review]

File: zhmcclient_mock/_urihandler.py [hunk 1]
```diff
@@ -3091,6 +3091,75 @@ def post(method, hmc, uri, uri_parms, body, logon_required,
...
+            raise new_exc  # zhmcclient_mock.InvalidResourceError
+        cpc = adapter.manager.parent
+        if not cpc.dpm_enabled:
+            raise CpcNotInDpmError(method, uri, cpc)
+
+        filter_args = query_parms
+        assigned_partitions = []
+        for partition in cpc.partitions.list(filter_args):
+            # Check if partition has a NIC backed by this adapter
+            for nic in partition.nics.list():
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply zhmcclient_mock/_urihandler.py:3123]: Very good catch. I was (too narrowly) looking only at network adapters...

[the remark this answers is not in the document]
Comment (andy-maier) [reply zhmcclient_mock/_urihandler.py]: I have added support for crypto cards and accelerator cards.

The support for ficon cards depends on the newly created issue #1557, the code to support them as been commented out again and marked with an according TODO.

The support for all other adapters has been marked as a TODO.

[the remark this answers is not in the document]
Comment (andy-maier) [reply zhmcclient_mock/_urihandler.py]: I have et this PR to "waiting", to be resumed after issue #1557 is resolved.

**Comment by andy-maier** [comment]
I have set this PR to "waiting", to be resumed after issue https://github.com/zhmcclient/python-zhmcclient/pull/1557 is resolved.

**Review by EdGue42 — APPROVED** [review]

**Comment by andy-maier** [comment]
Separated the enabling of the check for success from this PR into a separate one, so that this PR can pass and be merged.

**Pull request merged**